### PR TITLE
Add form for HTLC Withdraw

### PR DESCRIPTION
**Pull request by man-of-code**

### CODE REVIEW

**Comment by robert-zaremba** [comment]
Could you please copy-paste a screen shot of the new version?

**Comment by man-of-code** [comment]
@robert-zaremba Sorry for late response. This is how the updated UI looks like. 
image

**Comment by ratik21** [comment]
Can we also display escrow address & escrow balance before withdraw? What do you think?

**Comment by man-of-code** [comment]
> Can we also display escrow address & escrow balance before withdraw? What do you think?

@ratik21  We can show escrow address, but I didn't find any way to show escrow balance.

**Comment by ratik21** [comment]
> I didn't find any way to show escrow balance.

Please go through the js-sdk documentation. You can use `algodv2.accountInformation(address).do()`

**Comment by man-of-code** [comment]
@ratik21

**Comment by man-of-code** [comment]
@ratik21

image

File: htlc/src/components/signer.js [hunk 1]
```diff
@@ -97,47 +81,102 @@ const GetAccounts = () => {
...
+      let secret = document.getElementById("secret").value;
+      if (secret === "") {
+        return "Please enter secret(hash pre-image).";
+      }
+      let amount = document.getElementById("amount").value;
+      if (amount === "") {
+        return "Please enter amount.";
+      }
+      if (!Number(amount)) {
+        return "Entered amount is not a number.";
```
Comment (ratik21) [inline htlc/src/components/signer.js]: Amount can also be a `bigint`. SDK makeTransaction object also handles this.

File: htlc/src/components/withdrawHtlc.js [hunk 2]
```diff
@@ -10,26 +11,48 @@ const waitForConfirmation = async function (algodclient, txId) {
...
+export async function WithdrawHtlc(receiver, secret, amount) {
   try {
     const sender = YAMLData.compiledHash; //"YEUJZXDQAUA3J6WK3RZQJKHCM3S7E6UQF3I6AKRGTIGACFXDKFSKBYCYKU";
-    const receiver = prompt('Enter Receiver Address'); //"EDXG4GGBEHFLNX6A7FGT3F6Z3TQGIU6WVVJNOXGYLVNTLWDOCEJJ35LWJY";
-    const secret = prompt('Enter Secret'); //"hero wisdom green split loop element vote belt";
-    let amount = prompt('Enter Amount'); // 2e6
-    const secretBytes = new Uint8Array(Buffer.from(secret)); 
+    // const receiver = prompt('Enter Receiver Address'); //"EDXG4GGBEHFLNX6A7FGT3F6Z3TQGIU6WVVJNOXGYLVNTLWDOCEJJ35LWJY";
+    // const secret = prompt('Enter Secret'); //"hero wisdom green split loop element vote belt";
+    // let amount = prompt('Enter Amount'); // 2e6
```
Comment (ratik21) [inline htlc/src/components/withdrawHtlc.js]: Please remove these comments

File: htlc/src/components/withdrawHtlc.js [hunk 3]
```diff
@@ -38,17 +61,22 @@ async function WithdrawHtlc() {
     amount = amount - params.fee;
```
Comment (ratik21) [inline htlc/src/components/withdrawHtlc.js]: Why are we doing this?  I think that's why i am getting the error i posted above.

**Review by ratik21 — COMMENTED** [review]
Getting this error (params are correct)
image

File: htlc/src/components/signer.js (hunk 1 above)
Comment (man-of-code) [reply htlc/src/components/signer.js]: oh. okay

File: htlc/src/components/withdrawHtlc.js (hunk 2 above)
Comment (man-of-code) [reply htlc/src/components/withdrawHtlc.js]: Yes. I forgot to remove these.

File: htlc/src/components/withdrawHtlc.js (hunk 3 above)
Comment (man-of-code) [reply htlc/src/components/withdrawHtlc.js]: While sending the amount from escrow to reciever's account, a fee needs to be deducted from the escrow.

Comment (ratik21) [reply htlc/src/components/withdrawHtlc.js]: We don't need to explicitly deduct the fee here.

Comment (man-of-code) [reply htlc/src/components/withdrawHtlc.js]: Yes. I understand now. Making changes.

**Review by ratik21 — APPROVED** [review]

**Pull request merged**